From 8735c01239b5c060198a033641081bc8f63ae539 Mon Sep 17 00:00:00 2001
From: Wowee0 <co81407@coxj.com>
Date: Wed, 25 Dec 2019 19:00:30 -0600
Subject: [PATCH] UI issue wrt/removing first UTXO

---
 js/coinbin.js | 23 ++++++++++++++++++++---
 1 file changed, 20 insertions(+), 3 deletions(-)

diff --git a/js/coinbin.js b/js/coinbin.js
index ede27692..55749cc6 100644
--- a/js/coinbin.js
+++ b/js/coinbin.js
@@ -782,9 +782,25 @@ $(document).ready(function() {
 		window.location = "#fees";
 	});
 
+	// this is when the user clicks on the '-' button on the FIRST row of inputs only
+	// there is a different function for the '-' button on other rows
 	$(".txidClear").click(function(){
-		$("#inputs .row:first input").attr('disabled',false);
-		$("#inputs .row:first input").val("");
+		var numTxIns = $('#inputs div.row').length;
+		if (numTxIns > 1) {
+			// when there are subsequent rows of data we want the
+			// second row data moved into this row, and the second row deleted
+			// github coinbin/issues/132
+			$("#inputs .txId:first").val($("#inputs .txId")[1].value);
+			$("#inputs .txIdN:first").val($("#inputs .txIdN")[1].value);
+			$("#inputs .txIdScript:first").val($("#inputs .txIdScript")[1].value);
+			$("#inputs .txIdAmount:first").val($("#inputs .txIdAmount")[1].value);
+			$(this).parent().parent().next().fadeOut().remove(); // remove the second row
+		}
+		else {
+			$("#inputs .row:first input").val("");
+			// when clearing the first row, enable it so user can enter information manually
+			$("#inputs .row:first input").attr('disabled',false);
+		}
 		totalInputAmount();
 	});
 
@@ -900,7 +916,8 @@ $(document).ready(function() {
 		}
 
 		if($("#clearInputsOnLoad").is(":checked")){
-			$("#inputs .txidRemove, #inputs .txidClear").click();
+			$("#inputs .txidRemove").click();
+			$("#inputs .txidClear").click();
 		}
 
 		$("#redeemFromBtn").html("Please wait, loading...").attr('disabled',true);

