From f5c2f48efc3d2f4be11bc4c248e38a15fa3ae7fb Mon Sep 17 00:00:00 2001
From: Wowee0 <co81407@coxj.com>
Date: Wed, 12 Feb 2020 10:57:15 -0600
Subject: [PATCH] fix: some bech32 addresses were not recognised

---
 js/coin.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/coin.js b/js/coin.js
index 2c03e146..d6d1f67e 100644
--- a/js/coin.js
+++ b/js/coin.js
@@ -307,7 +307,7 @@
 
 				return o;
 			} else {
-				return false;
+				throw "Invalid checksum";
 			}
 		} catch(e) {
 			bech32rs = coinjs.bech32redeemscript(addr);

